src/sentry/api/serializers/models/projectcodeowners.py

```diff
@@ -1,9 +1,26 @@
 from sentry.api.serializers import Serializer, register
 from sentry.models import ProjectCodeOwners
+from sentry.utils.db import attach_foreignkey
 
 
 @register(ProjectCodeOwners)
 class ProjectCodeOwnersSerializer(Serializer):
+    def get_attrs(self, item_list, user, **kwargs):
+        attach_foreignkey(
+            item_list,
+            ProjectCodeOwners.repository_project_path_config,
+            related=("organization_integration",),
+        )
+
+        return {
+            item: {
+                "provider": item.repository_project_path_config.organization_integration.integration.provider
+                if item.repository_project_path_config.organization_integration
+                else "unknown"
+            }
+            for item in item_list
+        }
```

Comment:
You can actually access this property in `serialize` if you want, since it's already cached on the object by `attach_foreignkey`. It's fine here too